src/clj/witan/send/send.clj

```diff
@@ -216,37 +216,49 @@
-                                                                transition-matrix
-                                                                (ds/row-maps population))
-                                         (p/beta-params-joiners valid-states
-                                                                transition-matrix-filtered
-                                                                (ds/row-maps population)))
-   :leaver-beta-params (stitch-ay-state-params splice-ncy
-                                               (p/beta-params-leavers valid-states transition-matrix)
-                                               (p/beta-params-leavers valid-states transition-matrix-filtered))
-   :joiner-state-alphas (stitch-ay-params splice-ncy
-                                          (p/alpha-params-joiner-states valid-states (u/transitions-map transition-matrix))
-                                          (p/alpha-params-joiner-states valid-states (u/transitions-map transition-matrix-filtered)))
-   :projected-population (->> (ds/row-maps population)
-                              (group-by :calendar-year)
-                              (map (fn [[k v]] [k (u/total-by-academic-year v)]))
-                              (into {}))
-   :mover-beta-params (stitch-ay-state-params splice-ncy
-                                              (p/beta-params-movers valid-states transition-matrix)
-                                              (p/beta-params-movers valid-states transition-matrix-filtered))
-   :mover-state-alphas (stitch-ay-state-params splice-ncy
-                                               (p/alpha-params-movers valid-states transition-matrix)
-                                               (p/alpha-params-movers valid-states transition-matrix-filtered))
-   :setting-cost-lookup (->> (ds/row-maps setting-cost)
-                             (map (juxt (juxt :need :setting) :cost))
-                             (into {}))
-   :valid-setting-academic-years valid-setting-academic-years
-   :transition-matrix original-transitions
-   :population population})
+                   population valid-setting-academic-years original-transitions setting-cost
+                   filter-transitions-from]
+  (let [start-map {:population-by-age-state initial-state
+                   :valid-setting-academic-years valid-setting-academic-years
+                   :transition-matrix original-transitions
+                   :population population
+                   :setting-cost-lookup (->> (ds/row-maps setting-cost)
+                                             (map (juxt (juxt :need :setting) :cost))
+                                             (into {}))
+                   :projected-population (->> (ds/row-maps population)
+                                              (group-by :calendar-year)
+                                              (map (fn [[k v]] [k (u/total-by-academic-year v)]))
```

Comment:
Consider medley.core/map-vals

Reply:
not for just one call

Reply:
tested, wouldn't work in this scenario, as I'm changing the shape of the data